File: AlphaStandardDemo/Models/Safra.vb
Imports System.ComponentModel.DataAnnotations
Imports System.ComponentModel.DataAnnotations.Schema
Imports AlphaStandardDemo.Interfaces

Namespace Models
  <Table("safras", Schema:="cadastros")>
  Public Class Safra
    Implements IEntity

    <Key>
    Public Property Id As Integer Implements IEntity.Id

    <StringLength(100)>
    Public Property Descricao As String
    Public Property Inativo As Boolean?
    Public Property DataInicial As Date?
    Public Property DataFinal As Date?
  End Class
End Namespace

File: AlphaLib/Models/Safra.cs
using AlphaLib.Interfaces;
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace AlphaLib.Models
{
    [Table("safras", Schema = "cadastros")]
    public class Safra : IEntity
    {
        [Key]
        public int Id { get; set; }

        [StringLength(100)]
        public string Descricao { get; set; }
        public bool? Inativo { get; set; }
        public DateTime? DataInicial { get; set; }
        public DateTime? DataFinal { get; set; }
    }
}
